yoticlient.go

```diff
@@ -37,30 +43,55 @@ type Client struct {
 	Key []byte
 }
 
-// GetUserProfile requests information about a Yoti user using the one time use token generated by the Yoti login process.
+//ActivityDetails represents the result of an activity between a user and the application
+type ActivityDetails struct {
+	UserProfile Profile
+	// RememberMeID is a unique identifier Yoti assigns to your user, but only for your app
+	// if the same user logs into your app again, you get the same id
+	// if she/he logs into another application, Yoti will assign a different id for that app
+	RememberMeID string
+}
+
+// Deprecated: Will be removed in v3.0.0. Use `GetProfile` instead. GetUserProfile requests information about a Yoti user using the one time use token generated by the Yoti login process.
 // It returns the outcome of the request. If the request was successful it will include the users details, otherwise
 // it will specify a reason the request failed.
-func (client *Client) GetUserProfile(token string) (UserProfile, error) {
-	return getActivityDetails(doRequest, token, client.SdkID, client.Key)
+func (client *Client) GetUserProfile(token string) (userProfile UserProfile, firstError error) {
+	var errStrings []string
+	userProfile, _, errStrings = getActivityDetails(doRequest, token, client.SdkID, client.Key)
+	if len(errStrings) > 0 {
+		firstError = errors.New(errStrings[0])
+	}
+	return userProfile, firstError
 }
 
-func getActivityDetails(requester httpRequester, encryptedToken, sdkID string, keyBytes []byte) (result UserProfile, err error) {
+// GetActivityDetails requests information about a Yoti user using the one time use token generated by the Yoti login process.
+// It returns the outcome of the request. If the request was successful it will include the users details, otherwise
+// it will specify a reason the request failed.
+func (client *Client) GetActivityDetails(token string) (ActivityDetails, []string) {
+	_, activityDetails, errStrings := getActivityDetails(doRequest, token, client.SdkID, client.Key)
+	return activityDetails, errStrings
+}
+
+func getActivityDetails(requester httpRequester, encryptedToken, sdkID string, keyBytes []byte) (userProfile UserProfile, activityDetails ActivityDetails, errStrings []string) {
```

Comment:
@LKDraytonBrown & @pn-santos - we are thinking of removing this list of error strings returned (and also from the TimeAtttribute and JSONAttribute), and instead just log the errors, and return nil for the attributes that cannot be parsed. I know this is not the traditional Go way, in that we want to propagate any errors, but we don't want clients to fail to get any of the other attributes if there is a problem with one. 

Do you think this is an acceptable solution to not return the errors?

Reply:
Drayton suggested to tag @serverside, who might be able to answer this question

Reply:
So a `nil` attribute means there was an error processing that attribute?

If that is so, then I guess that is acceptable, though if there is a way for us to "categorize" the errors, it would be preferable to return a "partial success" data structure in which you have the ability to describe/identify the error for a particular attribute (even if for now there is only 1 error type). 

Returning `nil` (i.e. the equivalent of a "there was _an_ error processing attribute X") gives the client the ability to know there was an error, but that's it. Maybe that's enough, but consider whether it might be of use to the client to know the type of error (either now or in a hypothetical future). The `nil` approach will not allow you to "extend" the error information without a backwards incompatible change (if it ever becomes a necessity).

Reply:
This is a by product of having to use a generics alternative: 
It is indeed an accessor method, but we are having to wait until the last moment to convert the "Yoti(Attribute)" to a "Time(Attribute)", so it is not possible in the current format to retrieve it before this point. 

So you're saying it would be more appropriate to call them `GetDateOfBirth` and `GetStructuredPostalAddress` (or something similar), and return an error alongside? Or are these also accessor names that would suggest it wouldn't return an error, and it should be `ParseDateOfBirth` and `ParseStructuredPostalAddress`?